components/script/dom/blob.rs

```diff
@@ -114,13 +118,37 @@ impl Blob {
     }
 }
 
+/// The data required to initialize a new blob object from a serialized form.
+pub struct StorageKey {
+    pub index: u32,
+    pub name_space: u32,
+}
+
+impl ToSerializeOperations for StorageKey {
+    fn to_serialize_operations(&self) -> Vec<SerializeOperation> {
+        vec![SerializeOperation::Uint32Pair(self.name_space, self.index)]
+    }
+}
+
+#[allow(unsafe_code)]
+impl FromStructuredClone for StorageKey {
+    unsafe fn from_structured_clone(r: *mut JSStructuredCloneReader) -> StorageKey {
+        let mut key = StorageKey {
+            name_space: 0,
+            index: 0,
+        };
+        assert!(JS_ReadUint32Pair(r, &mut key.name_space, &mut key.index,));
+        key
+    }
+}
+
 impl Serializable for Blob {
+    type Data = StorageKey;
```

Comment:
By the way I just saw that there is an inconsistency in the current code between messageport and blob, and that we don't really need this key. We can just use the currently unused [data](https://github.com/servo/servo/blob/717baff17b651606e3e7bf53ba20dc00ae996a38/components/script/dom/bindings/structuredclone.rs#L119), just like [ports do](https://github.com/servo/servo/blob/717baff17b651606e3e7bf53ba20dc00ae996a38/components/script/dom/messageport.rs#L220)